core-c-bridge/include/temporal-sdk-core-c-bridge.h

```diff
@@ -629,6 +652,130 @@ typedef struct TemporalCoreWorkerReplayPushResult {
   const struct TemporalCoreByteArray *fail;
 } TemporalCoreWorkerReplayPushResult;
 
+/**
+ * Interop function pointer signature of the retry predicate of [CallbackBasedWorkerClientPollOptions]
+ */
+typedef bool (*TemporalCoreCallbackBasedWorkerClientNoRetryPredicate)(void *user_data,
+                                                                      enum TemporalCoreCallbackBasedWorkerClientGRPCCode code);
+
+/**
+ * Interop mirror of [temporal_sdk_core::PollOptions]
+ */
+typedef struct TemporalCoreCallbackBasedWorkerClientPollOptions {
+  /**
+   * always non-null (could be empty)
+   */
+  struct TemporalCoreByteArrayRef task_queue;
+  /**
+   * true if `no_retry_predicate` is valid
+   */
+  bool has_no_retry;
+  /**
+   * Decide short‐circuit GRPC error
+   */
+  TemporalCoreCallbackBasedWorkerClientNoRetryPredicate no_retry_predicate;
+  /**
+   * Opaque context pointer passed back to your predicate thunk
+   */
+  void *no_retry_user_data;
+  /**
+   * true if `timeout_override.is_some()`
+   */
+  bool has_timeout_ms;
+  /**
+   * timeout in milliseconds
+   */
+  uint64_t timeout_ms;
+} TemporalCoreCallbackBasedWorkerClientPollOptions;
+
+/**
+ * Function pointer for RPC callbacks delivering a `(success, failure)`
+ * pair as non-owning `ByteArrayRef`s to Rust.
+ */
+typedef void (*TemporalCoreWorkerClientCallbackTrampoline)(void *user_data,
+                                                           struct TemporalCoreByteArrayRef success,
+                                                           struct TemporalCoreByteArrayRef failure);
+
+/**
+ * Interop function pointer table implementing the RPC logic of the [temporal_sdk_core::WorkerClient] trait via callbacks.
+ */
+typedef struct TemporalCoreCallbackBasedWorkerClientRPCs {
+  void *user_data;
+  void (*poll_workflow_task)(void *user_data,
+                             struct TemporalCoreByteArrayRef request,
+                             const struct TemporalCoreCallbackBasedWorkerClientPollOptions *poll_opts,
+                             TemporalCoreWorkerClientCallbackTrampoline callback,
+                             void *callback_ud);
```

Comment:
```suggestion
                             const struct TemporalCoreCallbackBasedWorkerClientPollOptions *poll_options,
                             TemporalCoreWorkerClientCallbackTrampoline callback,
                             void *callback_user_data);
```

Pedantic, but would like not to abbreviate things that are exposed in the header